clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/DcosServerGroupNameResolver.groovy

```diff
@@ -40,11 +40,11 @@ class DcosServerGroupNameResolver extends AbstractServerGroupNameResolver {
     }
 
     def filteredApps = apps.findAll {
-      DcosSpinnakerAppId.parse(it.id).get().serverGroupName.cluster == Names.parseName(clusterName).cluster
+      DcosSpinnakerAppId.parse(it.id, true).get().serverGroupName.cluster == Names.parseName(clusterName).cluster
```

Comment:
We could introduce a DcosSpinnakerAppId.parseVerbose(...) method instead of a boolean argument to make it a bit clearer what it does. Don't know if it's worth investing the time into though.